packages/main/src/Slider.js

```diff
@@ -0,0 +1,167 @@
+import Float from "@ui5/webcomponents-base/dist/types/Float.js";
+import { fetchI18nBundle, getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import SliderBase from "./SliderBase.js";
+
+// Template
+import SliderTemplate from "./generated/templates/SliderTemplate.lit.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-slider",
+	languageAware: true,
+	managedSlots: true,
+	properties: /** @lends sap.ui.webcomponents.main.Slider.prototype */  {
+		value: {
+			type: Float,
+			defaultValue: 0,
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.Slider
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-slider
+ * @public
+ */
+class Slider extends SliderBase {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get template() {
+		return SliderTemplate;
+	}
+
+	static get styles() {
+		return SliderBase.styles;
+	}
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	onBeforeRendering() {
+		// Update initial Slider UI representation and normalize internal state
+		// Normalize Slider value according to min/max properties
+		// Normalize the step value and draw tickmarks/labels if specified
+		this._syncUIAndState();
```

Comment:
This is private and is defined within SliderBase.
I see it's been used in Base's onBeforeRendering. Why don't calling the overwritten method instead?

Reply:
I am thinking about making it protected as well as most of the shared private functions in SliderBase?